src/lib/structures/expr.ml

```diff
@@ -311,393 +311,374 @@ module F_Htbl : Hashtbl.S with type key = t =
       begin match ty with
         | Ty.Trecord { Ty.lbs = lbs; _ } ->
-          assert (List.length xs = List.length lbs);
-          fprintf fmt "{";
+          assert (List.compare_lengths xs lbs = 0);
+          Fmt.pf ppf "{";
           ignore (List.fold_left2 (fun first (field,_) e ->
-              fprintf fmt "%s%s = %a"  (if first then "" else "; ")
-                (Hstring.view field) print e;
+              Fmt.pf ppf "%s%s = %a"  (if first then "" else "; ")
+                (Hstring.view field) pp e;
               false
             ) true lbs xs);
-          fprintf fmt "}";
-        | _ -> assert false
+          Fmt.pf ppf "}";
+        | _ ->
+          (* Exclude by the typechecker. *)
+          assert false
       end
 
-    (* TODO: introduce PrefixOp in the future to simplify this ? *)
-    | Sy.Op op, [e1; e2] when op == Sy.Pow || op == Sy.Integer_round ||
-                              op == Sy.Max_real || op == Sy.Max_int ||
-                              op == Sy.Min_real || op == Sy.Min_int ->
-      fprintf fmt "%a(%a,%a)" Sy.print f print e1 print e2
+    | Sy.(Op ((Pow | Integer_round | Max_real | Min_real | Max_int
+              | Min_int) as op)), [e1; e2] ->
+      Fmt.pf ppf "%a(%a, %a)" pp_operator op pp e1 pp e2
 
-    (* TODO: introduce PrefixOp in the future to simplify this ? *)
-    | Sy.Op (Sy.Constr hs), ((_::_) as l) ->
-      fprintf fmt "%a(%a)"
-        Hstring.print hs (Util.print_list ~sep:"," ~pp:print) l
+    | Sy.(Op (Constr _ as op)), _::_ ->
+      Fmt.pf ppf "%a(%a)" pp_operator op Fmt.(list ~sep:comma pp) xs
 
-    | Sy.Op _, [e1; e2] ->
-      if Options.get_output_smtlib () then
-        fprintf fmt "(%a %a %a)" Sy.print f print e1 print e2
-      else
-        fprintf fmt "(%a %a %a)" print e1 Sy.print f print e2
-
-    | Sy.Op Sy.Destruct (hs, grded), [e] ->
-      fprintf fmt "%a#%s%a"
-        print e (if grded then "" else "!") Hstring.print hs
+    | Sy.(Op Destruct (hs, grded)), [e] ->
+      Fmt.pf ppf "%a#%s%a"
+        pp e (if grded then "" else "!") Hstring.print hs
 
+    | Sy.Op op, [e1; e2] ->
+      Fmt.pf ppf "(%a %a %a)" pp e1 pp_operator op pp e2
 
-    | Sy.In(lb, rb), [t] ->
-      fprintf fmt "(%a in %a, %a)" print t Sy.print_bound lb Sy.print_bound rb
-
+    | Sy.In (lb, rb), [t] ->
+      Fmt.pf ppf "(%a in %a, %a)" pp t Sy.print_bound lb Sy.print_bound rb
 
     | _, [] ->
-      fprintf fmt "%a" Sy.print f
+      Fmt.pf ppf "%a" Sy.print f
 
     | _, _ ->
-      fprintf fmt "%a(%a)" Sy.print f (Util.print_list ~sep:"," ~pp:print) xs
+      Fmt.pf ppf "%a(%a)" Sy.print f Fmt.(list ~sep:comma pp) xs
+
+  and pp_trigger ppf { content; _ } =
+    Fmt.list ~sep:Fmt.comma pp ppf content
 
-  and print_triggers fmt trs =
-    List.iter (fun { content = l; _ } ->
-        Format.fprintf fmt "| %a@," (Util.print_list ~sep:"," ~pp:print) l;
-      ) trs
+  and pp_triggers ppf trs =
+    Fmt.list ~sep:(fun ppf () -> Fmt.pf ppf "|@,") pp_trigger ppf trs
```

Comment:
Note that this changes the place where the cut occurs, did you test the output? I am not sure cutting immediately after the `|` will do what we want.

(As an aside: `fun ppf () -> Fmt.pf ppf "|@,"` is `Fmt.any "|@,"`)

Reply:
I tested and it wasn't pretty good. I fixed it.